nibabel/cifti2/cifti2.py

```diff
@@ -0,0 +1,848 @@
+import numpy as np
+
+from .. import xmlutils as xml
+from ..externals.six import string_types
+from ..externals.six.moves import reduce
+from ..filebasedimages import FileBasedHeader, FileBasedImage
+from ..nifti2 import Nifti2Image
+
+
+CIFTI_MAP_TYPES = ('CIFTI_INDEX_TYPE_BRAIN_MODELS',
+                   'CIFTI_INDEX_TYPE_PARCELS',
+                   'CIFTI_INDEX_TYPE_SERIES',
+                   'CIFTI_INDEX_TYPE_SCALARS',
+                   'CIFTI_INDEX_TYPE_LABELS')
+
+CIFTI_MODEL_TYPES = ('CIFTI_MODEL_TYPE_SURFACE',
+                     'CIFTI_MODEL_TYPE_VOXELS')
+
+CIFTI_SERIESUNIT_TYPES = ('SECOND',
+                          'HERTZ',
+                          'METER',
+                          'RADIAN')
+
+CIFTI_BrainStructures = ('CIFTI_STRUCTURE_ACCUMBENS_LEFT',
+                         'CIFTI_STRUCTURE_ACCUMBENS_RIGHT',
+                         'CIFTI_STRUCTURE_ALL_WHITE_MATTER',
+                         'CIFTI_STRUCTURE_ALL_GREY_MATTER',
+                         'CIFTI_STRUCTURE_AMYGDALA_LEFT',
+                         'CIFTI_STRUCTURE_AMYGDALA_RIGHT',
+                         'CIFTI_STRUCTURE_BRAIN_STEM',
+                         'CIFTI_STRUCTURE_CAUDATE_LEFT',
+                         'CIFTI_STRUCTURE_CAUDATE_RIGHT',
+                         'CIFTI_STRUCTURE_CEREBELLAR_WHITE_MATTER_LEFT',
+                         'CIFTI_STRUCTURE_CEREBELLAR_WHITE_MATTER_RIGHT',
+                         'CIFTI_STRUCTURE_CEREBELLUM',
+                         'CIFTI_STRUCTURE_CEREBELLUM_LEFT',
+                         'CIFTI_STRUCTURE_CEREBELLUM_RIGHT',
+                         'CIFTI_STRUCTURE_CEREBRAL_WHITE_MATTER_LEFT',
+                         'CIFTI_STRUCTURE_CEREBRAL_WHITE_MATTER_RIGHT',
+                         'CIFTI_STRUCTURE_CORTEX',
+                         'CIFTI_STRUCTURE_CORTEX_LEFT',
+                         'CIFTI_STRUCTURE_CORTEX_RIGHT',
+                         'CIFTI_STRUCTURE_DIENCEPHALON_VENTRAL_LEFT',
+                         'CIFTI_STRUCTURE_DIENCEPHALON_VENTRAL_RIGHT',
+                         'CIFTI_STRUCTURE_HIPPOCAMPUS_LEFT',
+                         'CIFTI_STRUCTURE_HIPPOCAMPUS_RIGHT',
+                         'CIFTI_STRUCTURE_OTHER',
+                         'CIFTI_STRUCTURE_OTHER_GREY_MATTER',
+                         'CIFTI_STRUCTURE_OTHER_WHITE_MATTER',
+                         'CIFTI_STRUCTURE_PALLIDUM_LEFT',
+                         'CIFTI_STRUCTURE_PALLIDUM_RIGHT',
+                         'CIFTI_STRUCTURE_PUTAMEN_LEFT',
+                         'CIFTI_STRUCTURE_PUTAMEN_RIGHT',
+                         'CIFTI_STRUCTURE_THALAMUS_LEFT',
+                         'CIFTI_STRUCTURE_THALAMUS_RIGHT')
+
+
+def _value_if_klass(val, klass, none_ok=True):
+    if none_ok and val is None:
+        return val
+    elif isinstance(val, klass):
+        return val
+    else:
+        raise ValueError('Not a valid %s instance.' % klass.__name__)
+
+
+def _value_or_make_klass(val, klass):
```

Comment:
You're not using this one anymore.  I see you've commented it out in various places - should it go?